gen.py

```diff
@@ -97,93 +101,119 @@ def convert_svg2png(infile, outfile, w, h):
 
         png_io = BytesIO()
         img.write_to_png(png_io)
-        with open(outfile, 'wb') as fout:
+        with open(outfile, "wb") as fout:
             fout.write(png_io.getvalue())
         svg.close()
         png_io.close()
         img.finish()
 
 
-# Only certain icon sizes may be covered
-try:
-    sizes = listdir("icons/{0}".format(theme))
-except FileNotFoundError:
-    exit("The theme {0} does not exists. Please reclone"
-         "the repository and try again.".format(theme))
-
-
-# The Generation Stuff
+# Android Generation Code
 if platform == "android":
     print("\nGenerating Android theme...")
-    theme_name = "com.numix.icons_{0}".format(theme)
+
+    # Define and create theme locations
+    theme_name = "com.numix.icons_" + theme
     android_dir = theme_name + "/MainActivity22/app/src/main/res/"
     app_filter = android_dir + "xml/appfilter.xml"
     theme_dir = android_dir + "drawable-xxhdpi/"
+    mkdir(theme_dir)
+    mkdir(path.dirname(app_filter))
 
+    # app_filter file contains Android meta data for the theme
     app_filter_content = '<?xml version="1.0" encoding="utf-8"?>\n'
     app_filter_content += '<resources>\n'
 
-    mkdir(theme_dir)
-    mkdir(path.dirname(app_filter))
-
     for icon_name, icon in icons.items():
         for component_info in icon.get("android", []):
+            # Check icon source exists before it's used
             source = "icons/{0}/48/{1}.svg".format(theme, format(icon_name))
+            if not path.exists(source):
+                continue
+
+            # Render icon to resources directory
             drawable_name = icon_name.replace(".", "_").replace("-", "_")
-            output = "{0}/{1}.png".format(theme_dir,
-                                          drawable_name)
-            if path.exists(source):
-                convert_svg2png(source, output, 192, 192)
-            app_filter_content += '\t<item component="ComponentInfo{' + \
-                component_info + '}" drawable="' + drawable_name + '"/>\n'
+            output = "{0}/{1}.png".format(theme_dir, drawable_name)
+            convert_svg2png(source, output, 192, 192)
+
+            # Add icons app_filter data (afd) to the app_filter file
+            afd = '\t<item component="ComponentInfo{{{0}}}" drawable="{1}"/>\n'
+            app_filter_content += afd.format(component_info, drawable_name)
```

Comment:
Cleaner representation of what this line actually looks like than splitting it as previous.